daemon/oci_linux.go

```diff
@@ -733,11 +733,19 @@ func (daemon *Daemon) createSpec(c *container.Container) (*specs.Spec, error) {
 	}
 
 	if apparmor.IsEnabled() {
-		appArmorProfile := "docker-default"
-		if len(c.AppArmorProfile) > 0 {
+		var appArmorProfile string
+		if c.AppArmorProfile != "" {
 			appArmorProfile = c.AppArmorProfile
 		} else if c.HostConfig.Privileged {
 			appArmorProfile = "unconfined"
+		} else {
+			// Unattended upgrades and other fun services can unload AppArmor
+			// profiles inadvertently. So in order to make sure that we keep
+			// the profile enabled, we will dynamically reload it if necessary.
+			if err := ensureDefaultAppArmorProfile(); err != nil {
```

Comment:
Do this later with ` if appArmorProfile == defaultApparmorProfile` condition to do it also loaded if `c.AppArmorProfile` equals `docker-default`.